WordPress/playground-tools
# Compress text files

## What?



Add compression to the Express server.

co-authored with @adamziel.

## Why?



Even when developing, it's important to be able to inspect the compressed size of the assets.

Other tools like `@wordpress/env` or Local also do it by default.

## How?



Add the [compression](https://www.npmjs.com/package/compression) middleware.

We had to add our own custom `filter` because the default one was not working with the `content-type` headers that WordPress returns. We added a very basic one, but it should be enough for development.

_The default one uses [this database of types](https://github.com/jshttp/mime-db/blob/master/db.json), but the [regular expressions](https://github.com/jshttp/compressible/blob/master/index.js) were failing._

## Testing Instructions






- Start the server.
- Check that the text files are compressed.

---

I couldn't get the tests to work. For some reason, the `fetch` is failing. I would appreciate some help with that.
